File: app/src/main/java/com/boost/espressotest/app/di/AppModule.java
package com.boost.espressotest.app.di;

import android.content.Context;

import com.boost.espressotest.app.EspressoTestApp;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * @author PerSpiKyliaTor on 19.01.18.
 */

@Module
public class AppModule {

    @Provides
    @Singleton
    Context provideContext(EspressoTestApp espressoTestApp) {
        return espressoTestApp.getApplicationContext();
    }
}

File: app/src/main/java/com/boost/espressotest/app/di/AppModule.kt
package com.boost.espressotest.app.di

import android.content.Context
import com.boost.espressotest.app.EspressoTestApp
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class AppModule {

    @Provides
    @Singleton
    internal fun provideContext(espressoTestApp: EspressoTestApp): Context {
        return espressoTestApp.applicationContext
    }
}
